src/main/java/net/malisis/core/client/gui/component/interaction/UISelect.java

```diff
@@ -446,7 +446,7 @@ public T select(Option<T> option)
 	 */
 	public T select(T obj)
 	{
-		return select(getOption(obj));
+		return (T) select(getOption(obj));
```

Comment:
Then the issue is on your end, because, my Eclipse, gradle and even Travis compile it fine.
Which makes sense too because select(Option) do return T already.
I configured Eclipse to remove unnecessary casts, and this one automatically goes away anyway.